Message/MessageFactory.php

```diff
@@ -88,6 +87,30 @@ protected function getDefaultOptions()
     }
 
     /**
+     * Find an email from database
+     *
+     * @param  string $reference
+     * @param  bool   $throwException
+     *
+     * @throws ReferenceNotFoundException
+     * @return  EmailInterface|null
+     */
+    public function getEmail($reference, $throwException = false)
+    {
+        if (!isset($this->emails[$reference])) {
+            $this->emails[$reference] = $this->em->getRepository($this->options['email_class'])->findOneByReference($reference);
+        }
+
+        $email = $this->emails[$reference];
+
+        if (true === $throwException) {
+            throw new ReferenceNotFoundException($reference, sprintf('Reference "%s" does not exist for email.', $reference));
```

Comment:
Why only `true === $throwException` ? if `$throwException` is set to `true`, even if `$email` is a valid object you will throw an exception.